core/templates/pages/exploration-editor-page/feedback-tab/thread-table/thread-table.component.spec.ts

```diff
@@ -12,50 +12,62 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
+import { TruncatePipe } from 'filters/string-utility-filters/truncate.pipe';
+import { DateTimeFormatService } from 'services/date-time-format.service';
+import { ThreadStatusDisplayService } from '../services/thread-status-display.service';
+import { ThreadTableComponent } from './thread-table.component';
+
 /**
  * @fileoverview Unit tests for threadTable.
  */
 
-describe('Thread table directive', function() {
-  var $scope = null;
-  var $uibModalInstance = null;
-  var DateTimeFormatService = null;
-
-  beforeEach(angular.mock.module('oppia'));
-  beforeEach(angular.mock.inject(function($injector, $componentController) {
-    var $rootScope = $injector.get('$rootScope');
-    DateTimeFormatService = $injector.get('DateTimeFormatService');
+export class MockDateTimeFormatService {
+  getLocaleAbbreviatedDatetimeString(): string {
+    return '11/21/2014';
+  }
+}
 
-    $uibModalInstance = jasmine.createSpyObj(
-      '$uibModalInstance', ['close', 'dismiss']);
+describe('Thread table component', () => {
+  let component : ThreadTableComponent;
+  let fixture: ComponentFixture<ThreadTableComponent>;
 
-    $scope = $rootScope.$new();
-    $componentController('threadTable', {
-      $scope: $scope,
-      $uibModalInstance: $uibModalInstance,
-    });
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      declarations: [ThreadTableComponent, TruncatePipe],
+      providers: [{ provide: DateTimeFormatService,
+        useClass: MockDateTimeFormatService }, ThreadStatusDisplayService]
+    }).compileComponents();
```

Comment:
Fix the indentation (can't edit properly on web)

```suggestion
      providers: [
      {
         provide: DateTimeFormatService,
         useClass: MockDateTimeFormatService
       }, 
       ThreadStatusDisplayService
    ]}).compileComponents();
```

Reply:
Done